bitsail-connectors/connector-kudu/src/main/java/com/bytedance/bitsail/connector/kudu/source/reader/KuduRowDeserializer.java

```diff
@@ -48,48 +52,53 @@ public Row convert(RowResult rowResult) {
-      case "INT32":
-      case "INT":     // INT_TYPE_INFO
-        return rowResult -> rowResult.getInt(columnName);
-      case "INT64":
-      case "LONG":    // LONG_TYPE_INFO
-        return rowResult -> rowResult.getLong(columnName);
-      case "DATE":    // SQL_DATE_TYPE_INFO
-        return rowResult -> rowResult.getDate(columnName);
-      case "TIMESTAMP":   // SQL_TIMESTAMP_TYPE_INFO
-        return rowResult -> rowResult.getTimestamp(columnName);
-      case "FLOAT":   // FLOAT_TYPE_INFO
-        return rowResult -> rowResult.getFloat(columnName);
-      case "DOUBLE":  // DOUBLE_TYPE_INFO
-        return rowResult -> rowResult.getDouble(columnName);
-      case "DECIMAL": // BIG_DECIMAL_TYPE_INFO
-        return rowResult -> rowResult.getDecimal(columnName);
-      case "VARCHAR": // STRING_TYPE_INFO
-        return rowResult -> rowResult.getVarchar(columnName);
-      case "STRING":  // STRING_TYPE_INFO
-        return rowResult -> rowResult.getString(columnName);
-      case "STRING_UTF8": // BINARY_TYPE_INFO
-      case "BINARY":      // BINARY_TYPE_INFO
-        return rowResult -> rowResult.getBinary(columnName).array();
-      default:
-        throw new BitSailException(KuduErrorCode.UNSUPPORTED_TYPE, "Type " + typeName + " is not supported");
+    Class<?> curClass = typeInfo.getTypeClass();
+    if (TypeInfos.BOOLEAN_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getBoolean(columnName);
+    }
+    if (TypeInfos.BYTE_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getByte(columnName);
+    }
+    if (TypeInfos.SHORT_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getShort(columnName);
+    }
+    if (TypeInfos.INT_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getInt(columnName);
+    }
+    if (TypeInfos.LONG_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getLong(columnName);
+    }
+    if (TypeInfos.SQL_DATE_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getDate(columnName);
+    }
+    if (TypeInfos.SQL_TIMESTAMP_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getTimestamp(columnName);
+    }
+    if (TypeInfos.FLOAT_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getFloat(columnName);
+    }
+    if (TypeInfos.DOUBLE_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getDouble(columnName);
+    }
+    if (TypeInfos.BIG_DECIMAL_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getDecimal(columnName);
+    }
+    if (TypeInfos.STRING_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getString(columnName);
+    }
+    if (BasicArrayTypeInfo.BINARY_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getBinary(columnName);
     }
```

Comment:
getBinary() method will return a `ByteBuffer` object. 
It's supposed to return a `byte[]` object.

Reply:
> getBinary() method will return a `ByteBuffer` object. It's supposed to return a `byte[]` object.

Thank you for you suggestion , I have fixed, please help me review again thx ! @BlockLiu